sql/moz-fx-data-shared-prod/subscription_platform/apple_subscriptions/view.sql

```diff
@@ -0,0 +1,251 @@
+      AND 1 = ROW_NUMBER() OVER (PARTITION BY original_transaction_id ORDER BY period_offset DESC)
+    ) AS in_billing_grace_period,
+  FROM
+    apple_iap_period_aggregates
+),
+apple_iap_trial_periods AS (
+  SELECT
+    original_transaction_id,
+    user_id,
+    purchase_date AS start_time,
+    expires_date AS end_time,
+  FROM
+    apple_iap_events
+  WHERE
+    offer_type = 1
+  QUALIFY
+    1 = ROW_NUMBER() OVER (PARTITION BY original_transaction_id, user_id ORDER BY expires_date DESC)
+)
+SELECT
+  periods.user_id AS customer_id,
+  periods.subscription_id,
+  NULLIF(periods.original_subscription_id, periods.subscription_id) AS original_subscription_id,
+  periods.product_id AS plan_id,
+  CASE
+    periods.status
+  WHEN
+    1
+  THEN
+    "active"
+  WHEN
+    2
+  THEN
+    "expired"
+  WHEN
+    3
+  THEN
+    "in billing retry period"
+  WHEN
+    4
+  THEN
+    "in billing grace period"
+  WHEN
+    5
+  THEN
+    "revoked"
+  END
+  AS status,
+  periods.verified_at AS event_timestamp,
+  IF(
+    periods.end_time <= trial_periods.end_time,
+    NULL,
+    COALESCE(trial_periods.end_time, periods.start_time)
+  ) AS subscription_start_date,
+  periods.original_purchase_date AS created,
+  trial_periods.start_time AS trial_start,
+  trial_periods.end_time AS trial_end,
+  periods.auto_renew_status = 0 AS cancel_at_period_end,
+  periods.end_time AS ended_at,
+  -- https://developer.apple.com/documentation/appstoreservernotifications/expirationintent
+  CASE
+  WHEN
+    periods.expiration_intent = 1
+  THEN
+    "Cancelled by Customer"
+  WHEN
+    periods.expiration_intent = 2
+  THEN
+    "Payment Failed"
+  WHEN
+    periods.expiration_intent = 3
+  THEN
+    "Price Change Not Approved by Customer"
+  WHEN
+    periods.expiration_intent = 4
+  THEN
+    "Product Unavailable at Renewal"
+  WHEN
+    periods.revocation_reason IS NOT NULL
+  THEN
+    "Refund"
+  END
+  AS ended_reason,
+  periods.user_id AS fxa_uid,
+  "Apple Store" AS provider,
+  (
+    CASE
+    WHEN
+      CONTAINS_SUBSTR(periods.product_id, ".1_month_subscription")
+    THEN
+      STRUCT("month" AS plan_interval, 1 AS plan_interval_count)
+    WHEN
+      CONTAINS_SUBSTR(periods.product_id, ".6_mo_subscription")
+    THEN
+      ("month", 6)
+    WHEN
+      CONTAINS_SUBSTR(periods.product_id, ".1_year_subscription")
+    THEN
+      ("year", 1)
+    ELSE
+      ERROR("subscription period not found: " || periods.product_id)
+    END
+  ).*,
+  "America/Los_Angeles" AS plan_interval_timezone,
+  periods.bundle_id AS product_id,
+  periods.in_billing_grace_period,
+  IF(
+    (periods.in_billing_grace_period AND periods.grace_period_expires_date > periods.end_time),
+    periods.grace_period_expires_date - periods.end_time,
+    INTERVAL 0 DAY
+  ) AS billing_grace_period,
+  periods.promotion_codes,
+FROM
+  apple_iap_enhanced_period_aggregates AS periods
+LEFT JOIN
+  apple_iap_trial_periods AS trial_periods
+USING
+  (original_transaction_id)
```

Comment:
This was causing a fan-out.
```suggestion
  (original_transaction_id, user_id)
```
(ditto in nonprod view)